### PR TITLE
Added random sort "easter egg" and added tooltip for the random sort

### PR DESCRIPTION
**Pull request by jmigual**
Hello,
I have added thhe random sort discussed in [bug #1447082](https://bugs.launchpad.net/mixxx/+bug/1447082)

### CODE REVIEW

**Comment by daschuer** [comment]
Thank you for the PR. It works :-)
On order to merge this, you need to become a contributor. 
Please sign: https://docs.google.com/a/mixxx.org/spreadsheet/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ

and comment here when done.

File: src/library/librarytablemodel.cpp [hunk 1]
```diff
@@ -101,3 +105,13 @@ TrackModel::CapabilitiesFlags LibraryTableModel::getCapabilities() const {
             | TRACKMODELCAPS_CLEAR_BEATS
             | TRACKMODELCAPS_RESETPLAYED;
 }
+
+void LibraryTableModel::setSort(int column, Qt::SortOrder order)
+{
+    BaseSqlTableModel::setSort(column, order);
+
+    // Random sort easter egg, only in library view
+    if (column == 1) {
```
Comment (daschuer) [inline src/library/librarytablemodel.cpp]: Use
fieldIndex(ColumnCache::COLUMN_LIBRARYTABLE_PREVIEW)
to make the code robust against future changes.

File: src/library/librarytablemodel.h [hunk 2]
```diff
@@ -16,6 +16,8 @@ class LibraryTableModel : public BaseSqlTableModel {
     int addTracks(const QModelIndex& index, const QList<QString>& locations);
     TrackModel::CapabilitiesFlags getCapabilities() const;
     static const QString DEFAULT_LIBRARYFILTER;
+
+    void setSort(int column, Qt::SortOrder order);
```
Comment (daschuer) [inline src/library/librarytablemodel.h]: you can use her the new c++11 override keyword.

File: src/library/librarytablemodel.cpp [hunk 3]
```diff
@@ -101,3 +105,13 @@ TrackModel::CapabilitiesFlags LibraryTableModel::getCapabilities() const {
             | TRACKMODELCAPS_CLEAR_BEATS
             | TRACKMODELCAPS_RESETPLAYED;
 }
+
+void LibraryTableModel::setSort(int column, Qt::SortOrder order)
+{
```
Comment (daschuer) [inline src/library/librarytablemodel.cpp]: nit (sorry): We place { always at the end of a line. 
https://www.mixxx.org/wiki/doku.php/coding_guidelines

**Comment by daschuer** [comment]
Some extra findings, but nothing that stops us from merging:

Playing with it the sorting arrow on the preview header feels misplaces since random has no reverse.

It would be cool to have a the soring hierarchy working with random. 
Use case. sort random and than sort by star rating. Result: random soring of all 5 star tracks.

**Comment by jmigual** [comment]
Ok, now I've made the changes and signed to become a contributor :)

**Comment by daschuer** [comment]
Thank you. Are you working on the hierarchical random sort or should we merge this as it is?

**Comment by jmigual** [comment]
I think we would better merge as it is, but we can add it to the wishlist.

**Pull request merged**